bluecellulab/stimulus/factory.py

```diff
@@ -687,3 +371,213 @@ def sinespec(
+            relative_skew: float = 0.5,
+            threshold_current: Optional[float] = None,
+            seed: Optional[int] = None
+    ) -> Stimulus:
+        """Creates a ShotNoise instance, either with an absolute amplitude or
+        relative to a threshold current.
+
+        Args:
+            duration: Duration of the stimulus (ms).
+            rate: Mean rate of synaptic events (Hz).
+            amp_mean: Mean amplitude of events (nA), used if provided.
+            amp_var: Variance in amplitude of events.
+            rise_time: Rise time of synaptic events (ms).
+            decay_time: Decay time of synaptic events (ms).
+            mean_percent: Mean current as a percentage of threshold current (used if amp_mean is None).
+            sd_percent: Standard deviation as a percentage of threshold current (used if amp_var is None).
+            relative_skew: Skew factor for the shot noise process (default: 0.5).
+            threshold_current: Reference threshold current for percentage-based calculation.
+            seed: Optional random seed for reproducibility.
+
+        Returns:
+            A `Stimulus` object that can be plotted and injected.
+
+        Notes:
+            - If `amp_mean` is provided, `mean_percent` is ignored.
+            - If `threshold_current` is not provided, threshold-based parameters cannot be used.
+        """
+        is_amplitude_based = amp_mean is not None and amp_var is not None
+        is_threshold_based = (
+            threshold_current is not None
+            and threshold_current != 0
+            and mean_percent is not None
+            and sd_percent is not None
+        )
+
+        if is_amplitude_based:
+            if is_threshold_based:
+                logger.info(
+                    "amp_mean, threshold_current, and mean_percent are all set in ShotNoise."
+                    " Using amp_mean and ignoring threshold-based parameters."
+                )
+            return ShotNoise.amplitude_based(
+                dt=self.dt,
+                duration=duration,
+                amp_mean=amp_mean,  # type: ignore[arg-type]
+                rate=rate,
+                amp_var=amp_var,  # type: ignore[arg-type]
+                rise_time=rise_time,
+                decay_time=decay_time,
+                seed=seed
+            )
+
+        if is_threshold_based:
+            return ShotNoise.threshold_based(
+                dt=self.dt,
+                duration=duration,
+                rise_time=rise_time,
+                decay_time=decay_time,
+                mean_percent=mean_percent,  # type: ignore[arg-type]
+                sd_percent=sd_percent,  # type: ignore[arg-type]
+                threshold_current=threshold_current,  # type: ignore[arg-type]
+                relative_skew=relative_skew,
+                seed=seed
+            )
+
+        raise TypeError("You must provide either `amp_mean` and `amp_var`, or `threshold_current` with percentage values.")
```

Comment:
same here, threshold_current and mean_percent and sd_percent to have complete requirements